### ISSUE TITLE
Use dict instead of OrderedDict

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#6566]
`dict` is ordered now in versions of python we support. And according to https://realpython.com/python-ordereddict/ - `dict` seems to generally outperform `OrderedDict` in CPU and memory, and I don't think we use any of specific to OrderedDict functionality.

```
(git)lena:~datalad/datalad-maint[bf-sve]git
$> git grep OrderedDict | wc -l 
64
``` 
I think it would not hurt to just switch to just use `dict`.

**Comment by adswa** [comment]
I believe I found one specific use of OrderedDict, namely ``popitem(last=False)``, used in ``status``:

```
datalad/core/local/status.py:635:        qdspath, qpaths = paths_by_ds.popitem(last=False)
```

The [standard ``dict``'s ``popitem`` method](https://docs.python.org/3/library/stdtypes.html#dict.popitem) doesn't take arguments, and returns items in LIFO order, but the ``last=False`` parameter in ``OrderedDict``s results in FIFO order.

**Comment by yarikoptic** [comment]
good catch! `popitem(last=False)` could be replaced with some helper like
```
k = next(iter(d.keys())); v = d.pop(k)
```
I guess. The key would be the first one.  If a singular use, could be just applied as above without carrying to create a helper.

**Comment by adswa** [comment]
done in https://github.com/datalad/datalad/pull/7174

**Comment by yarikoptic-gitmate** [comment]
Issue fixed in [`0.18.0`](https://github.com/datalad/datalad/releases/tag/0.18.0)
